### PR TITLE
[Aladdin] [Do Not Merge]  Test to merge Aladdin generated examples

### PR DESCRIPTION
**Pull request by haroldrandom**
Those changes in src/*/_help.py come from parsing [aladdin.py which provided by Aladdin team](https://github.com/Azure/azure-cli/pull/11505/files). 

Those changes are only part of examples in  aladdin.py not all thx content.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/monitor/_help.py [hunk 1]
```diff
@@ -728,6 +740,16 @@
   - name: Update autoscale settings. (autogenerated)
     text: az monitor autoscale update --count 3 --email-administrator true --enabled true --max-count 5 --min-count 2 --name MyAutoscaleSettings --resource-group MyResourceGroup --tags key[=value]
     crafted: true
+  - name: Update autoscale settings. (autogenerated)
+    text: |
+        az monitor autoscale update \\
+        --name MyAutoscaleSettings \\
+        --resource-group MyResourceGroup \\
+        --set retentionPolicy.days=365 \\
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/monitor/_help.py]: three same settings? Can we choose to drop some examples in this CI?

File: src/azure-cli/azure/cli/command_modules/network/_help.py [hunk 2]
```diff
@@ -3963,6 +4170,16 @@
...
 short-summary: Update a service endpoint policy definition.
+examples:
+  - name: Update a service endpoint policy definition. (autogenerated)
+    text: |
+        az network service-endpoint policy-definition update \\
+        --add communities='12076:5010' \\
+        --name MyServiceEndpointPolicyDefinition \\
+        --policy-name MyPolicy \\
+        --resource-group MyResourceGroup \\
+        --subscription MySubscription
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/network/_help.py]: usually, we don't need this one. @haroldrandom Do you think we need this argument or not?

Comment (haroldrandom) [reply src/azure-cli/azure/cli/command_modules/network/_help.py]: Which one?

File: aladdin_parser.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    yaml_example_key_written = False
+
+    # get preceding indent
+    preceding_indent, example_inner_indent = calculate_preceding_indent(raw_cli_help)
+    # print('preceding_indent =', len(preceding_indent))
+    # print('example_inner_indent =', len(example_inner_indent))
+
+    # append Aladdin added examples
+    for parameter_seq, examples in formatted_aladdin_examples.items():
+        if parameter_seq in formatted_cli_examples:
```
[the remark this answers is not in the document]
Comment (haroldrandom) [reply aladdin_parser.py]: What do you mean by update?

File: src/azure-cli/azure/cli/command_modules/acr/_help.py [hunk 4]
```diff
@@ -883,6 +883,10 @@
         az acr task update --image MyImage --name MyTask --registry MyRegistry \\
             --context https://github.com/Azure-Samples/acr-build-helloworld-node.git
     crafted: true
+  - name: Update a task for an Azure Container Registry. (autogenerated)
+    text: |
+        az acr task update --base-image-trigger-enabled true --name MyTask --registry MyRegistry
+    crafted: true
```
[the remark this answers is not in the document]
Comment (haroldrandom) [reply src/azure-cli/azure/cli/command_modules/acr/_help.py]: I identity one example by its options sequence. So here I think it's not duplicated.
Any sugguestion to deal with this?

File: aladdin_parser.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+def format_examples(cmd, examples):
+    formatted = defaultdict(list)
+
+    for example in examples:
+        text = example['text']
+        command_line = text[text.index(cmd):]
+
+        parameter_section = command_line[command_line.index(cmd) + len(cmd):]
+        parameter_section = shlex.split(parameter_section)
```
[the remark this answers is not in the document]
Comment (haroldrandom) [reply aladdin_parser.py]: Could you give a reference? I can't find `--metadata a=b c=d`

**Comment by jiasli** [comment]
Nice code! How about using regex to simplify the matching? https://regex101.com/r/NnxGR9/1

```regex
helps\['(.*?)'\] = \"\"\"\n(.*?)\n\"\"\"
```

File: src/azure-cli/azure/cli/command_modules/acr/_help.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/acr/_help.py]: oh no.  I mean the `crafted: true`. It's same with autogenerated.

File: aladdin_parser.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MyronFanQiu) [reply aladdin_parser.py]: In the future, the value may change or an example with super set of parameters appear.

File: src/azure-cli/azure/cli/command_modules/network/_help.py (hunk 2 above)
Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/_help.py]: yes. It's not related with your CI work. Just the content of examples.

File: aladdin_parser.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (MyronFanQiu) [reply aladdin_parser.py]: no. Currently we don't meet such situation. But I think we may need in the future.

File: src/azure-cli/azure/cli/command_modules/network/_help.py (hunk 2 above)
Comment (haroldrandom) [reply src/azure-cli/azure/cli/command_modules/network/_help.py]: Should I remove this argument `--subscription` when counting duplicated example via sorted arguments sequence?
Any shortcoming?

**Comment by haroldrandom** [comment]
> Nice code! How about using regex to simplify the matching? https://regex101.com/r/NnxGR9/1
> 
> ```regexp
> helps\['(.*?)'\] = \"\"\"\n(.*?)\n\"\"\"
> ```

Elegant method to grab command and its help content.

I thought about this way, but we still need to write the original content to file. Even though it can grab comand and its help content as easily as import statement. So I pick the dummy way to parse YAML content as it costs only one linear scan.

I removed some dummy code via RegEx to make code more clean.
Really got inspired by the RegEx way.

**Pull request closed without merging**